simplebgc/src/commands/read_params.rs

```diff
@@ -188,32 +180,36 @@ pub enum RcMixChannel {
     Yaw,
 }
 
-#[derive(BgcPayload, Copy, Clone, Debug, PartialEq)]
-pub struct RcMix(
-    #[kind(enumeration)]
-    #[name("")]
-    #[format(u8)]
-    pub RcMixRate,
-    #[kind(enumeration)]
-    #[name("")]
-    #[format(u8)]
-    pub RcMixChannel,
-);
-
-impl FromPrimitive for RcMix {
-    fn from_i64(n: i64) -> Option<Self> {
-        FromPrimitive::from_u8(n as u8)
-    }
-
-    fn from_u8(b: u8) -> Option<Self> {
-        Some(RcMix(
-            FromPrimitive::from_u8(b & 0b111111)?,
-            FromPrimitive::from_u8(b >> 5)?,
-        ))
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub struct RcMix {
+    pub rc_mix_rate: u8,
+    pub rc_mix_channel: RcMixChannel,
+}
+
+impl Payload for RcMix {
+    fn from_bytes(b: Bytes) -> Result<Self, PayloadParseError>
+    where
+            Self: Sized {
+        // We expect one byte
+        if b.len() < 1 {
+            return Err(PayloadParseError::InvalidFlags{ name: "RcMixRate".into() });
+        }
+        let byte = b[0];
+        let rc_mix_rate = 0b00011111 & byte;
+        let rc_mix_channel = RcMixChannel::from_u8(byte >> 5).unwrap();
```

Comment:
add a comment explaining that bits 0..5 are mix rate and 6..7 are mix channel. you can figure it out by reading the code, but just to reduce cognitive load

Reply:
Done